netket/hilbert/abstract_hilbert.py

```diff
@@ -59,124 +59,25 @@ def __init__(self):
-            i: The index of the desired site
-
-        Returns:
-            The number of degrees of freedom at that site
-        """
-        raise NotImplementedError()  # pragma: no cover
-
-    @abc.abstractmethod
-    def states_at_index(self, i: int) -> Optional[List[float]]:
-        r"""A list of discrete local quantum numbers at the site i.
-        If the local states are infinitely many, None is returned.
-
-        Args:
-            i: The index of the desired site.
-
-        Returns:
-            A list of values or None if there are infintely many.
-        """
-        raise NotImplementedError()  # pragma: no cover
-
-    def numbers_to_states(
-        self, numbers: Union[int, np.ndarray], out: Optional[np.ndarray] = None
-    ) -> np.ndarray:
-        r"""Returns the quantum numbers corresponding to the n-th basis state
-        for input n. n is an array of integer indices such that
-        :code:py:`numbers[k]=Index(states[k])`.
-        Throws an exception iff the space is not indexable.
-
-        Args:
-            numbers (numpy.array): Batch of input numbers to be converted into arrays of
-                quantum numbers.
-            out: Optional Array of quantum numbers corresponding to numbers.
-        """
-        if out is None:
-            out = np.empty((np.atleast_1d(numbers).shape[0], self.size))
-
-        if np.any(numbers >= self.n_states):
-            raise ValueError("numbers outside the range of allowed states")
-
-        if np.isscalar(numbers):
-            return self._numbers_to_states(np.atleast_1d(numbers), out=out)[0, :]
-        else:
-            return self._numbers_to_states(numbers, out=out)
-
-    def states_to_numbers(
-        self, states: np.ndarray, out: Optional[np.ndarray] = None
-    ) -> Union[int, np.ndarray]:
-        r"""Returns the basis state number corresponding to given quantum states.
-        The states are given in a batch, such that states[k] has shape (hilbert.size).
-        Throws an exception iff the space is not indexable.
-
-        Args:
-            states: Batch of states to be converted into the corresponding integers.
-            out: Array of integers such that out[k]=Index(states[k]).
-                 If None, memory is allocated.
-
-        Returns:
-            numpy.darray: Array of integers corresponding to out.
-        """
-        if states.shape[-1] != self.size:
-            raise ValueError(
-                f"Size of this state ({states.shape[-1]}) not"
-                f"corresponding to this hilbert space {self.size}"
-            )
-
-        states_r = np.asarray(np.reshape(states, (-1, states.shape[-1])))
-
-        if out is None:
-            out = np.empty(states_r.shape[:-1], dtype=np.int64)
-
-        out = self._states_to_numbers(states_r, out=out.reshape(-1))
-
-        if states.ndim == 1:
-            return out[0]
-        else:
-            return out.reshape(states.shape[:-1])
+        return False  # pragma: no cover
```

Comment:
Also probably better as `NotImplementedError`.

Reply:
i'd leave it as is, so that one not have to worry unless you know you are implementing a specific, finite hilbert spcae?

it seems to me a sane default

Reply:
uhm I agree that if you want to leave it maybe the best is to go for ```NotImplementedError```, I am not sure there is a general reason to assume that the default hilbert is not discrete?